pageserver/src/storage_sync2.rs

```diff
@@ -290,6 +291,10 @@ struct UploadQueueInitialized {
     /// in-progress and queued operations
     latest_files: HashMap<LayerFileName, LayerFileMetadata>,
 
+    /// How many file uploads or deletions been scheduled, since the
+    /// last (scheduling of) metadata index upload?
+    latest_files_changes: u64,
```

Comment:
We use `files` in storage_sync already (see `latest_files` above).
Let's stick to it.

But I think it would be good to include to make clear in the variable name that this counter refers to member `latest_files`.

```
/// How many file uploads or deletions been scheduled since the
/// last scheduling of metadata index upload.
latest_files_changes_since_metadata_upload_scheduled: u64
```

Reply:
> latest_files_changes_since_metadata_upload_scheduled

Ugh, that's horribly wrong. But actually I quite like it. It's descriptive, and given how we use that field, even though it's long it doesn't require reformatting any lines to accommodate it :-). So I went with that.